pepper-apis/studybuilder-cli/src/main/java/org/broadinstitute/ddp/studybuilder/task/SingularMedicalReleaseV2.java

```diff
@@ -0,0 +1,283 @@
+    // We will then create a new version of this activity.
+    // Important: to avoid clashing with existing question stable ids, in new file any question that was mean to
+    // be reused from original activity was named with _NEW prefix.
+    // Once old activity version has been updated so it is deprecated and new activity version has been stored
+    // we will update the new activity version to replace questions with stable id that ends with _NEW with original
+    // question.
+
+    @Override
+    public void run(Handle handle) {
+        activityDao = handle.attach(ActivityDao.class);
+        jdbiActVersion = handle.attach(JdbiActivityVersion.class);
+        jdbiActivity = handle.attach(JdbiActivity.class);
+        jdbiFormActivityFormSection = handle.attach(JdbiFormActivityFormSection.class);
+
+        LanguageStore.init(handle);
+
+        User adminUser = handle.attach(UserDao.class).findUserByGuid(studyCfg.getString("adminUser.guid")).get();
+        StudyDto studyDto = handle.attach(JdbiUmbrellaStudy.class).findByStudyGuid(studyCfg.getString("study.guid"));
+
+
+        String studyGuid = studyDto.getGuid();
+        long studyId = studyDto.getId();
+
+        String activityCode = dataCfg.getString("activityCode");
+        ActivityDto activityDto = jdbiActivity
+                .findActivityByStudyGuidAndCode(studyGuid, activityCode)
+                .orElseThrow(() -> new DDPException(
+                        "Could not find activity for activity code " + activityCode + " and study id " + studyGuid));
+        long activityId = activityDto.getActivityId();
+
+        ActivityVersionDto originalV1VersionDto = jdbiActVersion.findByActivityCodeAndVersionTag(studyDto.getId(),
+                activityCode, "v1").orElseThrow(DDPException::new);
+
+        ActivityVersionDto v2VersionDto = createNewVersion(activityId, activityCode, studyGuid, adminUser);
+
+        ActivityVersionDto updatedV1VersionDto = jdbiActVersion.findByActivityCodeAndVersionTag(studyDto.getId(),
+                activityCode, "v1").orElseThrow(DDPException::new);
+
+        // save before update for checking
+        List<FormSectionMembershipDto> v1SectionMembershipsBeforeVersionUpdate = jdbiFormActivityFormSection
+                .findOrderedSectionMemberships(activityId, originalV1VersionDto.getRevStart());
+
+        updateSectionVersion(activityId, updatedV1VersionDto);
+
+        List<FormSectionMembershipDto> v1SectionMembersAfterVersionUpdate = jdbiFormActivityFormSection
+                .findOrderedSectionMemberships(activityId, updatedV1VersionDto.getRevStart());
+        // Does section list look same before and after version update?
+        if (!extractSectionIds(v1SectionMembershipsBeforeVersionUpdate)
+                .equals(extractSectionIds(v1SectionMembersAfterVersionUpdate))) {
+            throw new DDPException("We messed up updating sections");
+        }
+
```

Comment:
I peppered the patch with these sort of checks as I wanted to be 100% sure as I modified stuff that I was not breaking existing stuff. There are more of these sort of inline tests below.